v2/fb-xcom-api-validator/src/encoding.ts

```diff
@@ -0,0 +1,71 @@
+import base58 from "bs58";
+import * as base32 from "hi-base32";
+
+export class UnsupportedEncodingFormatError extends Error {}
+
+export type Encoding = "plain" | "base64" | "hexstr" | "base32" | "base58";
+
+export abstract class Encoder {
+    public abstract encode(payload: string): string;
+    public abstract decode(payload: string): string;
+}
```

Comment:
This should be an interface, not an abstract class. Abstract classes are very very rare in TS and it should be a warning for you, if you find yourself creating one.